### PR TITLE
fix: update adapter ut (mock http requests)

### PR DESCRIPTION
**Pull request by zyyw**
Closes: #15318

Signed-off-by: Shengwen Yu <yshengwen@vmware.com>

### CODE REVIEW

File: src/pkg/reg/adapter/artifacthub/adapter_test.go [hunk 1]
```diff
@@ -1,12 +1,35 @@
...
+	ahRegistry := &model.Registry{
+		Type: model.RegistryTypeArtifactHub,
+		URL:  "https://artifacthub.io",
+	}
+	ahAdapter, err = newAdapter(ahRegistry)
+	if err != nil {
+		os.Exit(1)
+	}
+	gock.InterceptClient(ahAdapter.client.httpClient)
+}
```
Comment (heww) [inline src/pkg/reg/adapter/artifacthub/adapter_test.go:27]: Suggest making these as a function.

File: src/pkg/reg/adapter/dockerhub/adapter_test.go [hunk 2]
```diff
@@ -2,37 +2,44 @@ package dockerhub
...
-	assert.Nil(err)
-	assert.NotNil(adapter)
+	}
+	ad, err = newAdapter(r)
+	if err != nil {
+		os.Exit(1)
+	}
+	a := ad.(*adapter)
+	gock.InterceptClient(a.client.client)
+}
```
Comment (heww) [inline src/pkg/reg/adapter/dockerhub/adapter_test.go:39]: Revert these back to the function.

File: src/pkg/reg/adapter/artifacthub/adapter_test.go (hunk 1 above)
Comment (zyyw) [reply src/pkg/reg/adapter/artifacthub/adapter_test.go]: code has been updated per this review comment.

File: src/pkg/reg/adapter/dockerhub/adapter_test.go (hunk 2 above)
Comment (zyyw) [reply src/pkg/reg/adapter/dockerhub/adapter_test.go]: code updated

**Review by heww — APPROVED** [review]

**Review by bitsf — APPROVED** [review]

**Pull request merged**